src/hooks/connectWeb3/useWeb3React.tsx

```diff
@@ -77,25 +77,24 @@ export function ActiveWeb3Provider({
   children,
 }: PropsWithChildren<ActiveWeb3ProviderProps>) {
   const onError = console.error
+
   const [defaultChainId, setDefaultChainId] = useAtom(defaultChainIdAtom)
   useEffect(() => {
     if (propsDefaultChainId !== defaultChainId) setDefaultChainId(propsDefaultChainId)
   }, [propsDefaultChainId, defaultChainId, setDefaultChainId])
 
-  const integratorConnection = useMemo(() => getConnectionFromProvider(onError, provider), [onError, provider])
-  const metaMaskConnection = useMemo(
-    () => toWeb3Connection(initializeConnector<MetaMask>((actions) => new MetaMask({ actions, onError }))),
-    [onError]
+  // we're abusing useState a bit to ensure that `metaMaskConnection` is only ever computed once, i.e. is referentially static
+  const [metaMaskConnection] = useState(() =>
```

Comment:
This would more appropriately be `useMemo` - can we use `useMemo` and pass in `onError: console.error`?